PermissionsService/Services/PermissionsStore.cs

```diff
@@ -340,10 +375,26 @@ private async Task<string> FetchHttpSourceDocument(string sourceUri)
                 {
                     try
                     {
-                        var scopesForUrl = await GetScopesForRequestUrlAsync(request.RequestUrl, request.HttpMethod, scopeType);
-                        if (scopesForUrl == null || !scopesForUrl.Any())
-                            throw new InvalidOperationException($"Permissions information for '{request.HttpMethod} {request.RequestUrl}' was not found.");
+                        var resource = authZChecker.FindResource(request.RequestUrl);
 
+                        if (resource == null)
+                            throw new InvalidOperationException($"Permissions information for '{request.HttpMethod} {request.RequestUrl}' was not found.");
+                        var leastPrivilege = resource.FetchLeastPrivilege(request.HttpMethod);
+                        leastPrivilege.TryGetValue(request.HttpMethod, out var methodPermissions);
+                        methodPermissions.TryGetValue(scopeType.ToString(), out var leastPrivilegedPermissions);
```

Comment:
We should probably confirm that the `TryGetValue` succeeded to avoid null reference errors
```suggestion
                        if (leastPrivilege.TryGetValue(request.HttpMethod, out var methodPermissions) 
                            && methodPermissions.TryGetValue(scopeType.ToString(), out var leastPrivilegedPermissions))
```